src/App.jsx

```diff
@@ -1,13 +1,110 @@
-import { useState } from 'react'
-import './App.css'
+import React, { useState } from "react";
+import ClassCheckBox from "./components/ClassCheckbox";
+import classes from "./constants";
+import "./App.css";
 
-function App() {
+
+export default function App() {
+	const [data, setData] = useState(classes);
+
+	const handleClassCheck = (classIndex) => {
+		let dataClone = [...data];
+		let targetClass = dataClone[classIndex];
+
+		targetClass.checked = !targetClass.checked;
+		targetClass.sections = targetClass.sections.map((section) => {
+			let students = section.students.map((student) => ({
+				...student,
+				checked: targetClass.checked,
+			}));
+
+			return {
+				...section,
+				checked: targetClass.checked,
+				students,
+			};
+		});
+
+		dataClone[classIndex] = targetClass;
+		setData([...dataClone]);
+	};
+
+	const handleSectionCheck = (sectionIndex, classIndex) => {
+		let dataClone = [...data];
+		let sections = dataClone[classIndex].sections.map((section, index) => {
+			if (index === sectionIndex) {
+				section.checked = !section.checked;
+				let students = section.students.map((student) => ({
+					...student,
+					checked: section.checked,
+				}));
+
+				return {
+					...section,
+					students,
+				};
+			}
+			return section;
+		});
+
+		dataClone = handleAllSectionsCheck(dataClone, classIndex);
+		dataClone[classIndex].sections = sections;
+
+		setData([...dataClone]);
+	};
+
+	const handleStudentCheck = (studentIndex, sectionIndex, classIndex) => {
+		let dataClone = [...data];
+		let sections = dataClone[classIndex].sections;
+		let students = sections[sectionIndex].students.map((student, index) => {
+			if (index === studentIndex) {
+				return {
+					...student,
+					checked: !student.checked,
+				};
+			}
+			return student;
+		});
+
+		let allStudentsChecked = students.every((student) => student.checked === true);
+
+		sections[sectionIndex].students = students;
+		sections[sectionIndex].checked = allStudentsChecked;
+
+		dataClone[classIndex].sections = sections;
+
+		dataClone = handleAllSectionsCheck(dataClone, classIndex);
+
+		setData([...dataClone]);
+	};
+
+	const handleAllSectionsCheck = (dataClone, classIndex) => {
+		let allSectionChecked = dataClone[classIndex].sections.every(
+			(section) => section.checked === true
+		);
+		dataClone[classIndex].checked = allSectionChecked;
+
+		return dataClone;
+	};
 
 	return (
-		<>
-			<h1>Checkbox App</h1>
-		</>
-	)
+		<div className="App">
+			<h1>Checkbox React App</h1>
+			{data.map((classObj, classIndex) => (
```

Comment:
Same here also. Please add data check.

`{data?.map((classObj, classIndex) => (`

Reply:
Also what is data? Please use proper variable names. 
data and classObj do not match.

Reply:
Sure will make the changes.